woocommerce/woocommerce
# Include download IDs in csv import/export to retain import download permissions.

### All Submissions:






### Changes proposed in this Pull Request:

This PR updates the product export to include download IDs so that when importing and updating products, existing download permissions remain mapped appropriately (as the download ID does not get updated).

Closes #29686.

### How to test the changes in this Pull Request:

1. Apply this PR. Create a product that is downloadable and virtual, and setup all the required fields (price, etc).
2. Add at least 1 downloadable file to the product.
3. Save the product, then add it to your cart, and complete checkout.
4. (Optional: Note the download ID from the products' `_downloadable_files` meta value).
4. Export products.
5. Import the CSV file you just generated, but select "Existing products that match by ID or SKU will be updated. Products that do not exist will be skipped."
6. Verify that you're still able to download the file (that download permissions are retained).
7.  (Optional: Verify that the ID from `_downloadable_files` meta value hasn't changed).

### Other information:




### Changelog entry

> Fix - Download IDs are included in export CSV and imported when updating existing products to maintain download permissions.
